chain-crypto/src/algorithms/vrf/dleq.rs

```diff
@@ -11,26 +9,26 @@ pub struct Proof {
     z: Scalar,
 }
 
-const PROOF_SIZE: usize = 64; // Scalar is 32 bytes
-
 impl Proof {
+    pub const PROOF_SIZE: usize = Scalar::BYTES_LEN * 2;
+
     pub fn to_bytes(&self, output: &mut [u8]) {
-        assert_eq!(output.len(), PROOF_SIZE);
-        output[0..32].copy_from_slice(self.c.0.as_bytes());
-        output[32..64].copy_from_slice(self.z.as_bytes());
+        assert_eq!(output.len(), Self::PROOF_SIZE);
+        output[0..Scalar::BYTES_LEN].copy_from_slice(&self.c.0.to_bytes());
+        output[Scalar::BYTES_LEN..].copy_from_slice(&self.z.to_bytes());
     }
 
     pub fn from_bytes(slice: &[u8]) -> Option<Self> {
-        if slice.len() != PROOF_SIZE {
+        if slice.len() != Self::PROOF_SIZE {
             return None;
         }
-        let mut c_array = [0u8; 32];
-        c_array.copy_from_slice(&slice[0..32]);
-        let c = Scalar::from_canonical_bytes(c_array)?;
+        let mut c_array = [0u8; Scalar::BYTES_LEN];
+        c_array.copy_from_slice(&slice[0..Scalar::BYTES_LEN]);
+        let c = Scalar::from_bytes(&c_array)?;
 
-        let mut z_array = [0u8; 32];
-        z_array.copy_from_slice(&slice[32..64]);
-        let z = Scalar::from_canonical_bytes(z_array)?;
```

Comment:
The old implementation was enforcing the use of canonical bytes but the p256k1 backend implementation of `Scalar::from_bytes` does not, is this of concern?

Reply:
p256k1 also returns an error if the scalar is non-canonical (higher than the group order). You can find it [here](https://github.com/vincenthz/eccoxide/blob/master/src/curve/fiat/field_macros.rs#L404)